packages/cli/src/commands/hydrogen/env/list.ts

```diff
@@ -0,0 +1,108 @@
+import {Flags} from '@oclif/core';
+import Command from '@shopify/cli-kit/node/base-command';
+import {renderConfirmationPrompt, renderTable} from '@shopify/cli-kit/node/ui';
+import {outputContent, outputToken} from '@shopify/cli-kit/node/output';
+
+import {linkStorefront} from '../link.js';
+import {adminRequest} from '../../../lib/graphql.js';
+import {commonFlags} from '../../../lib/flags.js';
+import {getHydrogenShop} from '../../../lib/shop.js';
+import {getAdminSession} from '../../../lib/admin-session.js';
+import {
+  ListEnvironmentsQuery,
+  ListEnvironmentsSchema,
+} from '../../../lib/graphql/admin/list-environments.js';
+import {getConfig} from '../../../lib/shopify-config.js';
+import {logMissingStorefront} from '../../../lib/missing-storefront.js';
+import {logMissingLink} from '../../../lib/missing-link.js';
+
+export default class List extends Command {
+  static description = 'List the environments on your Hydrogen storefront.';
+
+  static hidden = true;
+
+  static flags = {
+    path: commonFlags.path,
+    shop: commonFlags.shop,
+  };
+
+  async run(): Promise<void> {
+    const {flags} = await this.parse(List);
+    await listEnvironments(flags);
+  }
+}
+
+interface Flags {
+  path?: string;
+  shop?: string;
+}
+
+export async function listEnvironments({path, shop: flagShop}: Flags) {
+  const shop = await getHydrogenShop({path, shop: flagShop});
+  const adminSession = await getAdminSession(shop);
+  const actualPath = path ?? process.cwd();
+  let configStorefront = (await getConfig(actualPath)).storefront;
+
+  if (!configStorefront?.id) {
+    logMissingLink({adminSession});
+
+    const runLink = await renderConfirmationPrompt({
+      message: outputContent`Run ${outputToken.genericShellCommand(
+        `npx shopify hydrogen link`,
+      )}?`.value,
+    });
+
+    if (!runLink) {
+      return;
+    }
+
+    await linkStorefront({path, shop: flagShop, silent: true});
+  }
+
+  configStorefront = (await getConfig(actualPath)).storefront;
+
+  if (!configStorefront) {
+    return;
+  }
+
+  const result: ListEnvironmentsSchema = await adminRequest(
+    ListEnvironmentsQuery,
+    adminSession,
+    {
+      id: configStorefront.id,
+    },
+  );
+
+  const hydrogenStorefront = result.hydrogenStorefront;
+
+  if (!hydrogenStorefront) {
+    logMissingStorefront({adminSession, storefront: configStorefront});
+    return;
+  }
+
+  const rows = hydrogenStorefront.environments
+    .filter(({type}) => type !== 'PREVIEW')
+    .map(({branch, name, url, type}) => ({
+      name,
+      branch,
+      url: type === 'PRODUCTION' ? hydrogenStorefront.productionUrl : url,
+    }));
+
+  renderTable({
```

Comment:
Maybe add an empty line `console.log('')` before this?

<img width="352" alt="image" src="https://github.com/Shopify/hydrogen/assets/1634092/7c16ab4f-0736-4b1d-879b-70e4d8f43c7f">